sql/core/src/test/scala/org/apache/spark/sql/DataFrameRangeSuite.scala

```diff
@@ -127,4 +133,28 @@ class DataFrameRangeSuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("Cancelling stage in a query with Range.") {
+    val listener = new SparkListener {
+      override def onTaskStart(taskStart: SparkListenerTaskStart): Unit = {
+        Thread.sleep(100)
```

Comment:
The only other thing in the SparkListener interface that seems to fit here seems to be `onExecutorMetricsUpdate()`. However, the metrics are send with the heartbeats, and the heartbeats are send around 10s apart. Hence, using this method makes the tests much slower (4s vs 20s).